edgePy/io/_io.py

```diff
@@ -7,15 +7,60 @@
 
 
 __all__ = [
-    'Importer',
-    'get_dataset_path'
+    'GroupImporter',
+    'DataImporter',
+    'get_dataset_path',
+    'get_open_function'
 ]
 
 
-class Importer(object):
+def get_open_function(filename):
+    decode_required = False
+    if filename.endswith("gz"):
+        open_function = gzip.open
+        decode_required = True
+    else:
+        open_function = open
+
+    return open_function, decode_required
+
+
+class GroupImporter(object):
+
+    def __init__(self, filename=None):
+        self.filename = str(filename) if filename else None
+        if not self.filename:
+            raise Exception("No group filename")
+        self.samples = {}
+        self.groups = {}
+        self.read_file()
+
+    def read_file(self):
+        open_function, decode_required = get_open_function(filename=self.filename)
+        with open_function(self.filename) as f:
+            for line in f:
+                if decode_required:
```

Comment:
You can drop the `decode_required` from the entire PR if you just use a `try-except` would be the Pythonic method for duck-typing:

```python
try:
    line = line.decode('utf8').strip().split(':')
except AttributeError:
    line = line.strip().split(':')
```